### PR TITLE
Add current package selection syntax

### LINKED ISSUE #6891: [CT-2059] [Feature] Syntax to restrict selection to the current package
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

There are many use cases for not running the models in your packages.

- Feeding a `dbt ls` of the models in the project to an external tool
- Only checking the build of the current project in a mesh-like structure
- Many others but I won't belabor it unless you disagree :)

Unfortunately there's no easy way that I'm aware of that you can filter to just the models in the project without hardcoding the selects or excludes to various projects. Particularly as the mesh grows, I can see this becoming more of an issue, and a desire to 'just run this project's models' becoming really helpful.

I would love to implement something like:
`dbt build --select this.*` or perhaps `dbt ls --exclude external` or `dbt run --internal-only`/`dbt run -i` -- i haven't thought of the ideal syntax, but hopefully you get the idea.

### Describe alternatives you've considered

You can exclude all the packages by name in YAML selectors, or you can `[project_name].*` but this isn't very convenient, particularly when working programmatically on top of the CLI, such as with editor extensions that would need to parse the project YAML to get the package name.

### Who will this benefit?

This will primarily benefit people using lots of Fivetran packages, a multi-project mesh structure, or some other configuration that involves lots of models contributed by packages that you don't necessarily want to constantly re-run when you're developing.

### Are you interested in contributing this feature?

Yes! Very much. Dying to contribute to the CLI in some way.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by barton996**
resolves #6891

### Problem
There are many use cases where it is useful to exclude models not in your package from your run selection.

There is no easy way to limit your selection to just models in your project without resorting to hardcoding in package names.

### Solution
This change adds 'this' as a valid package selector, enabling statements like `dbt ls -s package:this`

Running the selector above will return just the models from the current project.

### User documentation

User docs for this feature should be added here:
https://docs.getdbt.com/reference/node-selection/methods#the-package-method

The examples can be updated to look like this:
```shell
# These three selectors are equivalent
dbt run --select "package:snowplow"
dbt run --select "snowplow"
dbt run --select "snowplow.*"

dbt run --select "package:this"  # only run models in the current dbt project 
```

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [x] I have run this code in development and it appears to resolve the stated issue  
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has already received feedback and approval from Product or DX within [#6891](https://github.com/dbt-labs/dbt-core/issues/6891)
- [x] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions

### CODE REVIEW

**Comment by barton996** [comment]
@dbeatty10 @graciegoheen 
image

File: core/dbt/graph/selector_methods.py [hunk 1]
```diff
@@ -438,7 +438,10 @@ class PackageSelectorMethod(SelectorMethod):
     def search(self, included_nodes: Set[UniqueId], selector: str) -> Iterator[UniqueId]:
         """Yields nodes from included that have the specified package"""
         for node, real_node in self.all_nodes(included_nodes):
-            if fnmatch(real_node.package_name, selector):
+            if selector == "." and self.manifest.metadata.project_name is not None:
+                if fnmatch(real_node.package_name, self.manifest.metadata.project_name):
+                    yield node
+            elif fnmatch(real_node.package_name, selector):
```
[the remark this answers is not in the document]
Comment (barton996) [reply core/dbt/graph/selector_methods.py]: @dbeatty10 I buy it- implemented

File: core/dbt/graph/selector_methods.py [hunk 2]
```diff
@@ -437,6 +437,10 @@ def search(self, included_nodes: Set[UniqueId], selector: str) -> Iterator[Uniqu
 class PackageSelectorMethod(SelectorMethod):
     def search(self, included_nodes: Set[UniqueId], selector: str) -> Iterator[UniqueId]:
         """Yields nodes from included that have the specified package"""
+        # `.` is an alias for the current dbt project name
+        if selector == "." and self.manifest.metadata.project_name is not None:
```
Comment (jtcohen6) [inline core/dbt/graph/selector_methods.py:441]:
```suggestion
        # `this` is an alias for the current dbt project name
        if selector == "this" and self.manifest.metadata.project_name is not None:
```

**Review by jtcohen6 — COMMENTED** [review]
@barton996 Thank you for this PR! :)

@aranke and I revisited the question of what syntax makes most sense here, and after some more discussion, we think we prefer `package:this` over `package:.` (despite what I said a year ago). If you're amenable to that change, let's make it happen.

**Review by dbeatty10 — APPROVED** [review]

**Pull request merged**